packages/augur-core/source/contracts/trading/ZeroXTrade.sol

```diff
@@ -367,16 +450,66 @@ contract ZeroXTrade is Initializable, IZeroXTrade, IERC1155 {
         }
     }
 
-    /// @dev Decode ERC-1155 asset data from the format described in the AssetProxy contract specification.
+    /// @dev Decode MultiAsset asset data from the format described in the AssetProxy contract specification.
     /// @param _assetData AssetProxy-compliant asset data describing an ERC-1155 set of assets.
-    /// @return The ERC-1155 AssetProxy identifier, the address of the ERC-1155
+    /// @return The ERC-1155 AssetProxy identifier, the address of this ERC-1155
     /// contract hosting the assets, an array of the identifiers of the
     /// assets to be traded, an array of asset amounts to be traded, and
     /// callback data.  Each element of the arrays corresponds to the
     /// same-indexed element of the other array.  Return values specified as
     /// `memory` are returned as pointers to locations within the memory of
     /// the input parameter `assetData`.
     function decodeAssetData(bytes memory _assetData)
+        public
+        view
+        returns (
+            bytes4 _assetProxyId,
+            address _tokenAddress,
+            uint256[] memory _tokenIds,
+            uint256[] memory _tokenValues,
+            bytes memory _callbackData,
+            address _kycToken
+        )
+    {
+         // Read the bytes4 from array memory
+        assembly {
+            _assetProxyId := mload(add(_assetData, 32))
```

Comment:
It's not a problem, but this is a little funny. We are using the memory allocated for the return of the ERC1155 _PROXY_ID to check the `MULTI_ASSET_PROXY_ID`, which is then ignored due to the explicit return.

Probably the most code-efficient way to do it, but a little odd